xrpl/models/required.py

```diff
@@ -4,6 +4,8 @@
 __init__ as kwargs.
 """
 
+from typing import Any
+
 from typing_extensions import Final
 
-REQUIRED: Final[object] = object()
+REQUIRED: Final[Any] = object()  # noqa: ANN401
```

Comment:
As per my understanding, we have shifted the `# type: ignore` mypy directive into this line with a `# noqa: ANN401` directive. `mypy` will be unhappy with our code either way.

It would be more elegant to adopt a template with required+optional fields for every transaction and ledger-entry. The format used by rippled is instructive in this regard: https://github.com/XRPLF/rippled/blob/0ec17b6026298dc150099b66a2cecad0fe561d1b/include/xrpl/protocol/STValidation.h#L222

If you know of any other ideas to enforce a schema in Python, please share it.

However, that requires development effort. I'm happy with this change.

Reply:
> As per my understanding, we have shifted the `# type: ignore` mypy directive into this line with a `# noqa: ANN401` directive. `mypy` will be unhappy with our code either way.

`noqa` is ignoring a linter error, not mypy. mypy is perfectly happy with this situation.

> It would be more elegant to adopt a template with required+optional fields for every transaction and ledger-entry. The format used by rippled is instructive in this regard: https://github.com/XRPLF/rippled/blob/0ec17b6026298dc150099b66a2cecad0fe561d1b/include/xrpl/protocol/STValidation.h#L222
> 
> If you know of any other ideas to enforce a schema in Python, please share it.

I don't believe that's possible without radically changing our model setup.

The reason we have `REQUIRED` in the first place is because Python inheritance won't let us have required fields after optional fields. Since there are optional fields in the base `Transaction` class, it isn't happy if we then try to add required fields in child transaction type classes.

Further, if it ain't broke, don't fix it - this system works just fine, so I wouldn't think it's much of a priority (if at all) to mess with it, unless there are other benefits to be had from that.

Reply:
I agree with the spirit of your argument